animation_nodes/nodes/mesh/combine_mesh_data.py

```diff
@@ -36,7 +36,10 @@ def execute(self, vertexLocations, edgeIndices, polygonIndices, materialIndices)
             materialIndices = LongList(length = len(polygonIndices))
             materialIndices.fill(0)
         try:
-            return Mesh(vertexLocations, edgeIndices, polygonIndices,
-                    materialIndices, skipValidation = self.skipValidation)
+            mesh = Mesh(vertexLocations, edgeIndices, polygonIndices, skipValidation = self.skipValidation)
+            mesh.insertAttribute("Material Indices", AttributeType["MATERIAL_INDEX"], AttributeDomain["FACE"],
+                                  AttributeDataType["INT"], materialIndices)
```

Comment:
> Also, it might be a good idea to make `insertAttribute` take an attribute object instead of take the individual elements.

Omar, the *Insert Custom Attribute* node operates on the mesh but not an object.

Reply:
@3DSinghVFX I meant a python object. What I was saying is that instead of passing those individual values, construct an Attribute object from them and pass that object. So it should have the following interface:
```
insertAttribute(attribute: Attribute)
```

Reply:
Okay.